Update Upgrading guide for OpenVox (#4769)
theforeman/foreman-documentation: guides/common/modules/proc_upgrade-puppet-agent-7-to-openvox-agent-8.adoc

:_mod-docs-content-type: PROCEDURE

[id="upgrade-puppet-agent-7-to-openvox-agent-8"]
= Upgrade Puppet agent 7 to OpenVox agent 8

[role="_abstract"]

> If you use Puppet for host management, you have the option to upgrade `puppet agent 7 to OpenVox agent 8.

Reviewer: ```suggestion
If you use Puppet to configure hosts, you have the option to upgrade `puppet agent 7 to OpenVox agent 8.
```

Puppet agent 7 remains compatible with OpenVox in {Project} as a legacy alternative, but {Team} recommends to upgrade to OpenVox agent 8.

> For more information, see {ManagingConfigurationsPuppetDocURL}openvox-versions-supported-for-integration-with-satellite[OpenVox versions supported for integration with {Project}] in _{ManagingConfigurationsPuppetDocTitle}_.

Reviewer 2: ```suggestion
If you use Puppet to configure hosts, you can upgrade Puppet agent 7 to OpenVox agent 8.

Puppet agent 7 remains compatible with OpenVox in {Project} as a legacy alternative, but {Team} recommends to upgrade to OpenVox agent 8.
```

Author: @aneta-petrova I am guessing this was supposed to be for line 7 and 8?

Reviewer 2: Oh yes, it was. Not sure what happened.

> For more information, see {ManagingConfigurationsPuppetDocURL}openvox-versions-supported-for-integration-with-satellite[OpenVox versions supported for integration with Satellite] in _{ManagingConfigurationsPuppetDocTitle}_.

Reviewer: ```suggestion
For more information, see {ManagingConfigurationsPuppetDocURL}openvox-versions-supported-for-integration-with-satellite[OpenVox versions supported for integration with {Project}] in _{ManagingConfigurationsPuppetDocTitle}_.
```

:_mod-docs-content-type: PROCEDURE

[id="upgrade-puppet-agent-7-to-openvox-agent-8"]
= Upgrade Puppet agent 7 to OpenVox agent 8

[role="_abstract"]
If you use Puppet to configure hosts, you can upgrade Puppet agent 7 to OpenVox agent 8.

Puppet agent 7 remains compatible with OpenVox in {Project} as a legacy alternative, but {Team} recommends to upgrade to OpenVox agent 8.

> For more information, see {ManagingConfigurationsPuppetDocURL}openvox-versions-supported-for-integration-with-satellite[OpenVox versions supported for integration with {Project}] in _{ManagingConfigurationsPuppetDocTitle}_.

Reviewer: ```suggestion
For more information, see {ManagingConfigurationsPuppetDocURL}openvox-versions-supported-for-integration-with-{project-context}[OpenVox versions supported for integration with {Project}] in _{ManagingConfigurationsPuppetDocTitle}_.
```

.Prerequisites
* *{project-client-name} OpenVox* repository is enabled on the host.

> .Procedure

Reviewer 3: Note a blocker, but this is a very manual way. Shouldn't we call out that you can also use REX to do it in bulk? I'm struggling to find any good existing documentation to do it but manually doing this for 100s or 1000s of servers isn't really a good experience.

Note that you upgrade all packages (without specifying a specific name) then it'll also be upgraded.

Reviewer 2: There is a precedent for how to refer to rex jobs directly in this guide, actually: https://github.com/theforeman/foreman-documentation/blob/3cc18a41069c1166fa7892f35bf6f0311172a4e4/guides/common/modules/proc_upgrading-smartproxy-server.adoc?plain=1#L14-L20  Perhaps if we knew the right rex job to call, we could add a similar admonition?

* On your hosts, upgrade the `puppet-agent` agent package to `openvox-agent`:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----

> # {project-package-update} puppet-agent

Reviewer 3: You can't use this: `{project-package-update}` is only for Satellite and Capsule machines. It needs to be `client-package-update-el8`:
```suggestion
# {client-package-update-el8} puppet-agent
```

Users can also run `project-package-update` without specifying any package.

:_mod-docs-content-type: PROCEDURE

[id="upgrade-puppet-agent-7-to-openvox-agent-8"]
= Upgrade Puppet agent 7 to OpenVox agent 8

[role="_abstract"]
If you use Puppet to configure hosts, you can upgrade Puppet agent 7 to OpenVox agent 8.

Puppet agent 7 remains compatible with OpenVox in {Project} as a legacy alternative, but {Team} recommends to upgrade to OpenVox agent 8.
For more information, see {ManagingConfigurationsPuppetDocURL}openvox-versions-supported-for-integration-with-{project-context}[OpenVox versions supported for integration with {Project}] in _{ManagingConfigurationsPuppetDocTitle}_.

.Prerequisites
* *{project-client-name} OpenVox* repository is enabled on the host.

.Procedure
* On your hosts, upgrade the `puppet-agent` agent package to `openvox-agent`:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# {client-package-update-el8} puppet-agent

> ----

Reviewer: ```suggestion
----
+
```

You can also use remote execution to upgrade the package on multiple hosts.

> For more information, see {ManagingHostsDocURL}Upgrading_a_package_managing-hosts[Upgrading a package] in _{ManagingHostsDocTitle}_.

Reviewer 3: I had a look at that procedure to, but it's not really what I had in mind. That is a procedure to update a package for a single host, but IMHO you really want to do a bulk action. That usually starts with a host group since they often share content views. However, I couldn't find a procedure that actually does that.
